ivy_tests/test_ivy/test_frontends/test_jax/test_devicearray.py

```diff
@@ -90,6 +90,41 @@ def test_jax_devicearray_property_at(x_y_index):
     assert x.at[idx].get() == x[idx]
 
 
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="jax.numpy.array",
+    method_name="astype",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=["int32"],
```

Comment:
Can we make testing more comprehensive and generate castable dtypes based on Jax promotion rules
https://jax.readthedocs.io/en/latest/type_promotion.html